File: phonebook.py
def solution(phone_book):
    for v in phone_book:
        for iv in phone_book:
            if v == iv : continue
            if v.startswith(iv) : return False

    return True

#
# t1 = ["12232332", "12", "222222"]
# print(solution(t1))
# t2 = ["911", "97625999", "91125426"]
# print(solution(t2))
t3 = ["113", "12340", "123440", "12345", "98346"]
print(solution(t3))

File: Phonebook.java
import java.util.HashMap;

class Phonebook {
    public boolean solution(String[] phone_book) {
        for (String v : phone_book) {
            for(String iv : phone_book){
                if(iv.equals(v))continue;
                if(v.startsWith(iv)){
                    return false;
                }
            }
        }
        return true;
    }

    public static void main(String[] args) {
        Phonebook p = new Phonebook();
        String[] n1 = {"119", "97674223", "1195524421"};
        System.out.println(p.solution(n1));
        String[] n2 = {"123", "456", "789"};
        System.out.println(p.solution(n2));
        String[] n3 = {"12", "123", "1235", "567", "88"};
        System.out.println(p.solution(n3));
    }
}